### PR TITLE
task: fix the incomplete/wrong description of JOIN_WAKER bit.

### PR DESCRIPTION
**Pull request by agayev**

### CODE REVIEW

File: tokio/src/runtime/task/mod.rs [hunk 1]
```diff
@@ -71,10 +72,33 @@
...
+//!       to gain exclusive access to the waker field per rule 2, (ii) install a
+//!       waker, and (iii) successfully set JOIN_WAKER to one.
+//!
+//!    5. The JoinHandle can only change JOIN_WAKER if COMPLETE is zero (i.e.
+//!       the task hasn't yet completed). Hence, steps (i) or (iii) of rule 4
+//!       may fail due to a race.
+//!
+//!    6. Once COMPLETE is one (i.e. task has completed), the JoinHandle will
+//!       not modify JOIN_WAKER and the waker field. At this point the runtime
+//!       invokes the waker if there is one.
```
Comment (Darksonn) [inline tokio/src/runtime/task/mod.rs:101]: The join handle _can_ still modify the waker field when COMPLETE is one. This happens if COMPLETE is changed to one during step 4. (ii).

File: tokio/src/runtime/task/mod.rs [hunk 2]
```diff
@@ -71,10 +72,35 @@
...
+//!
+//!    5. The JoinHandle can change JOIN_WAKER only if COMPLETE is zero (i.e.
+//!       the task hasn't yet completed).
+//!
+//!    Rule 5 implies that the steps (i) or (iii) of rule 4 may fail due to a
+//!    race. If step (i) fails, then the attempt to install a waker is aborted.
+//!    If step (iii) fails because COMPLETE is set to one by another thread
+//!    after step (i), then the waker field is cleared. Once COMPLETE is one
+//!    (i.e. task has completed), the JoinHandle will not modify JOIN_WAKER.
+//!    After COMPLETE is one, the runtime invokes the waker if there is one.
```
Comment (Darksonn) [inline tokio/src/runtime/task/mod.rs:103]:
```suggestion
//!    After COMPLETE is set to one, the runtime invokes the waker if there is one.
```

**Review by Darksonn — APPROVED** [review]
Looks reasonable.

File: tokio/src/runtime/task/mod.rs [hunk 3]
```diff
@@ -71,10 +72,35 @@
...
+//!    a waker to the waker field. The JOIN_WAKER bit ensures safe access by
+//!    multiple threads to the waker field using the following rules:
+//!
+//!    1. JOIN_WAKER is initialized to zero.
+//!
+//!    2. If JOIN_WAKER is zero, then the JoinHandle has exclusive (mutable)
+//!       access to the waker field.
+//!
+//!    3. If JOIN_WAKER is one, then the JoinHandle and the runtime have shared
+//!       (read-only) access to the waker field.
```
Comment (Darksonn) [inline tokio/src/runtime/task/mod.rs:88]: The runtime only has read-only access when COMPLETE is one. If you don't do this, then the JoinHandle is taking away permissions of the runtime in step 4 (i), which is normally not allowed in rule-sets for atomics.

Comment (agayev) [reply tokio/src/runtime/task/mod.rs]: Fixed.

**Review by Darksonn — APPROVED** [review]

**Pull request merged**